File: backups/pre_ai_bundle_resume_20260609T220028/backend_app/api/v1/endpoints/expert_reviews.py
from datetime import date

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.deps import require_roles
from app.models.enums import UserRole
from app.models.user import User
from app.schemas.review import (
    ExpertReviewAction,
    ExpertReviewActionResult,
    ExpertReviewDetail,
    ExpertReviewQueueItem,
    ExpertReviewStats,
)
from app.services.expert_review_service import ExpertReviewService

router = APIRouter(prefix="/expert-reviews", tags=["expert-reviews"])


@router.get("", response_model=list[ExpertReviewQueueItem])
def list_review_queue(
    risk_level: str | None = None,
    status_filter: str | None = Query(default=None, alias="status"),
    organization_id: int | None = None,
    prescription_type: str | None = None,
    abnormal_feedback: bool | None = None,
    start_date: date | None = None,
    end_date: date | None = None,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    return ExpertReviewService(db).list_queue(
        risk_level=risk_level,
        status_filter=status_filter,
        organization_id=organization_id,
        prescription_type=prescription_type,
        abnormal_feedback=abnormal_feedback,
        start_date=start_date,
        end_date=end_date,
        current_user=current_user,
    )


@router.get("/stats", response_model=ExpertReviewStats)
def get_review_stats(
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    return ExpertReviewService(db).stats(current_user=current_user)


@router.get("/{prescription_id}", response_model=ExpertReviewDetail)
def get_review_detail(
    prescription_id: int,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    return ExpertReviewService(db).detail(prescription_id, current_user=current_user)


@router.post("/{prescription_id}/start", response_model=ExpertReviewActionResult)
def start_review(
    prescription_id: int,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).start_review(
        prescription_id=prescription_id,
        current_user=current_user,
    )
    return {"status": "IN_REVIEW", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/approve", response_model=ExpertReviewActionResult)
def approve_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).approve(
        prescription_id=prescription_id,
        current_user=current_user,
        review_comment=payload.review_comment,
        edited_prescription=payload.edited_prescription,
    )
    return {"status": "APPROVED", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/reject", response_model=ExpertReviewActionResult)
def reject_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).reject(
        prescription_id=prescription_id,
        current_user=current_user,
        review_comment=payload.review_comment,
    )
    return {"status": "REJECTED", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/refer", response_model=ExpertReviewActionResult)
def refer_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).refer(
        prescription_id=prescription_id,
        current_user=current_user,
        review_comment=payload.review_comment,
    )
    return {"status": "REFERRED", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/request-info", response_model=ExpertReviewActionResult)
def request_more_information(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).request_info(
        prescription_id=prescription_id,
        current_user=current_user,
        review_comment=payload.review_comment,
    )
    return {"status": "NEEDS_INFO", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/pause", response_model=ExpertReviewActionResult)
def pause_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).pause(
        prescription_id=prescription_id,
        current_user=current_user,
        review_comment=payload.review_comment,
    )
    return {"status": "PAUSED", "review": review, "prescription": prescription}

File: backups/completion_trend_chart_20260609T165830/frontend/src/api/expertReviews.ts
import { apiClient } from "./client";

export interface ReviewQueueItem {
  prescription_id: number;
  user_id: number;
  organization_id: number | null;
  risk_level: string;
  status: string;
  prescription_type: string;
  abnormal_feedback_count: number;
  version: number;
  created_at: string;
  review_id: number | null;
}

export interface ReviewQueueFilters {
  risk_level?: string;
  status?: string;
  organization_id?: number;
  prescription_type?: string;
  abnormal_feedback?: boolean;
  start_date?: string;
  end_date?: string;
}

export interface ReviewStats {
  average_review_hours: number;
  r2_pending_count: number;
  timeout_count: number;
}

export interface ReviewDetail {
  prescription: {
    id: number;
    risk_level: string;
    status: string;
    fitt_vp: Record<string, unknown> | null;
    precautions: string[];
    contraindications: string[];
    reassessment?: string;
    safety_notice?: string | null;
    evidence_refs: Array<Record<string, unknown>>;
  };
  review: {
    id: number;
    status: string;
    review_comment?: string | null;
    edited_prescription?: Record<string, unknown> | null;
  };
  health_snapshot: Record<string, Record<string, unknown> | null>;
  risk_rules: Array<Record<string, unknown>>;
  evidence_refs: Array<Record<string, unknown>>;
  template: Record<string, unknown> | null;
  candidate_actions: Array<Record<string, unknown>>;
  versions?: Array<Record<string, unknown>>;
  trends?: Record<string, unknown>;
}

export function listReviewQueue(filters: ReviewQueueFilters = {}) {
  return apiClient.get<ReviewQueueItem[]>("/expert-reviews", { params: filters }).then((response) => response.data);
}

export function getReviewStats() {
  return apiClient.get<ReviewStats>("/expert-reviews/stats").then((response) => response.data);
}

export function getReviewDetail(prescriptionId: number) {
  return apiClient.get<ReviewDetail>(`/expert-reviews/${prescriptionId}`).then((response) => response.data);
}

export function approvePrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/approve`, payload).then((response) => response.data);
}

export function startReview(prescriptionId: number) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/start`).then((response) => response.data);
}

export function rejectPrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/reject`, payload).then((response) => response.data);
}

export function referPrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/refer`, payload).then((response) => response.data);
}

export function requestMoreInformation(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/request-info`, payload).then((response) => response.data);
}

export function pausePrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/pause`, payload).then((response) => response.data);
}
